src/components/checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,153 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import styled from '@emotion/styled';
+import * as RadixCheckbox from '@radix-ui/react-checkbox';
+import { useCallback } from 'react';
+import { Spinner } from '../spinner';
+import { CheckboxIcon } from './CheckboxIcon';
+
+interface Props {
+  checked: boolean | 'indeterminate';
```

Comment:
```suggestion
  isChecked: boolean | 'indeterminate';
```

Reply:
I'm not sure about this one, since it isn't a pure boolean with the 'indeterminate' state...
What's your opinion there @jeremy-davis-sonarsource ?

Reply:
Yeah, it's not a boolean. `checked` is fine